### DISCUSSION TITLE
optional argument that only has one option

### DISCUSSION THREAD

**Discussion by 0withoutface0** [Rapptz/discord.py#9724] [General] [answered]
How can i make an optional argument that can be selected but the user doesnt have to chose True or False bc it will become True by default?
I dont want the argument to have options. 
![kép](https://github.com/Rapptz/discord.py/assets/97801778/7755a6dd-4e96-4630-ae7c-375c5a8e67e6)
something like this:
`async def dice(self, context, hidden: True = False):`
so by default hidden is False, but if the user selects hidden it becomes True

**Comment by AbstractUmbra** [comment]
You'd want the annotation to be `hidden: bool = True`
Tnis should achieve what you're aiming for.

**Reply by 0withoutface0** [reply]
no because the user then still have to chose from True or False options

**Reply by 0withoutface0** [reply]
the argument should not need any additional switches bc its obvious that it hides the message. the message is not hidden by default, but when the hidden argument is added after the command it should mean True by itself.

**Reply by Rapptz** [reply]
The option in the screenshot is the `bool` type hint. The `= True` bit makes it so it has a default value so it's not required. I am unsure what part of this doesn't answer your request. Have you actually tried it?

**Reply by AbstractUmbra** [reply]
Are you sure you have synced your CommandTree since this change? I just tested this:-

```py
    @app_commands.command()
    async def boolean(self, interaction: Interaction, hidden: bool = True) -> None:
        return await interaction.response.send_message(f"You have entered {hidden}!")

```
![DiscordCanary_zBhWX0gFyq](https://github.com/Rapptz/discord.py/assets/16031716/304d6c9a-fcb8-4d42-9a22-b0065e32ac33)

With `hidden` being an optional parameter:-
![MbbGYECQxF](https://github.com/Rapptz/discord.py/assets/16031716/7a43b38c-4a6b-4515-94d0-e099e07ea5e0)

**Reply by 0withoutface0** [reply]
I will reply in 40 minutes bc I just had to leave home. 
I have tested it. I have a running bot and it's doing `hidden: bool = False` just fine. But it asks the user to chose from True, False. I want it so that when someone adds the attribute it will mean True by default so the user doesn't have to define it.

**Reply by Rapptz** [reply, marked as answer]
Then what you're asking isn't possible. Discord has no such thing. I'm not even sure you can have choices with only one element.

**Reply by 0withoutface0** [reply]
Thank you for telling me :D I would've searched for a method for hours just to find out that its not possible.
I hope discord will add supoort for it bc i think it would be a useful feature.
